game/state/savemanager.cpp

```diff
@@ -163,23 +163,25 @@ bool writeArchiveWithBackup(SerializationArchive *archive, const UString &path,
 
 bool SaveManager::findFreePath(UString &path, const UString &name) const
 {
-	path = createSavePath("save_" + name);
-	if (fs::exists(path))
-	{
-		for (int retries = 5; retries > 0; retries--)
-		{
-			path = createSavePath("save_" + name + std::to_string(rand()));
-			if (!fs::exists(path))
-			{
-				return true;
-			}
-		}
+	path = createSavePath(name);
+	bool pathExists = fs::exists(path);
 
-		LogError("Unable to generate filename for save %s", name);
-		return false;
+	return !pathExists;
+}
+
+sp<SaveMetadata> SaveManager::getSaveGameIfExists(const UString &name) const
```

Comment:
Not related to your change, as it's just following what's already there, but I really need to remove the sp<> wrapper here - it's not *really* a thing that needs shared ownership and arguably a bit unnecessary.